### ISSUE TITLE
Invalid french accents in sentence collector data

### ISSUE THREAD

**Issue by lissyx** [mozilla/voice-web#2259]
We got a report of "words cut" being displayed (likely a font rendering issue). Further investigation lead to unexpected french `é` character: `é`.

A quick check on the data reveals:
```
$ git grep -c "é" server/data/fr/
server/data/fr/sentence-collector.json:1
server/data/fr/sentence-collector.txt:50800
```

**Comment by lissyx** [comment]
Proper fix is indeed using unicode NFKC normalization instead of NFKD: https://github.com/Common-Voice/commonvoice-fr/pull/78

**Comment by lissyx** [comment]
@MichaelKohler So, I've merged the fix on our repo, and regenerated data. How can we perform the fix on CommonVoice and/or Sentence Collector side ?

**Comment by lissyx** [comment]
@MichaelKohler: this is the list of deleted sentences [deleted.txt](https://github.com/mozilla/voice-web/files/3780148/deleted.txt)

**Comment by MichaelKohler** [comment]
## Sentence Collector
The following was found and deleted in the sentence collector:

[removed_sc.txt](https://github.com/mozilla/voice-web/files/3790958/removed_sc.txt)

This includes sentences that were never approved and therefore didn't get exported, but were found in the list lissyx provided.

## Export

The following sentences would be removed if we ran the export:

[removed_export.txt](https://github.com/mozilla/voice-web/files/3790960/removed_export.txt)

## Difference

There is a difference of 413 sentences (counting total lines of both files). These are sentences that were in the sentence collector but never were approved, or even were rejected. I haven't found a good way to get a real diff. I will keep looking. Once we have the list of these 413 sentences, I can cross-check them against my saved file for French to see how many of those should be re-add through the Sentence Collector. I will only re-add ones that were not already rejected anyway.

## Common Voice Database

Now, as far as I know, if we just do a new PR for https://github.com/mozilla/voice-web/tree/master/server/data/fr, already recorded sentences would be marked as invalid. This is not something we want here, as it's purely an encoding issue. 

My plan would be the following:

* We run the export of sentences from the Sentence Collector - this deletes quite a lot of the sentences in the sentence-collector.txt file
* We re-add the already approved, corrected files (see first section of this comment) as a new file in the `data/fr` folder.
* We somehow change the Common Voice database with the same transformations so once these two PRs are merged, the next deployment of Common Voice would basically just ignore it.
* We probably would need to correct the mapping between already recorded sentences and their corresponding sentence in the dataset - I don't know if that's how it works or not though..

@rileyjshaw hi Riley, can you give me some background on how we can avoid invalidating sentences as mentioned right above here? That would be great! :) Feel free to ping me in Slack for more information.

Once all of that is done, subsequent exports from the Sentence Collector will not do any changes for these sentences.

**Comment by MichaelKohler** [comment]
> @rileyjshaw hi Riley, can you give me some background on how we can avoid invalidating sentences as mentioned right above here? That would be great! :) Feel free to ping me in Slack for more information.

@rileyjshaw quick ping here :)

**Comment by rileyjshaw** [comment]
Hi @MichaelKohler, thanks for the ping – I missed the initial notification.

@phirework is the current lead on Common Voice, so I'm going to redirect this to her. Happy to join a discussion if it would help!

**Comment by phirework** [comment]
Thanks @rileyjshaw! 

@MichaelKohler, this is on my radar to investigate and I need to confer with some of the original devs on the project. I will be sure to ping you on Slack once I have a better idea of how to move forward.

**Comment by MichaelKohler** [comment]
Thanks, happy to provide more information if needed, feel free to ping me anytime

**Comment by MichaelKohler** [comment]
> There is a difference of 413 sentences (counting total lines of both files). These are sentences that were in the sentence collector but never were approved, or even were rejected. I haven't found a good way to get a real diff. I will keep looking. Once we have the list of these 413 sentences, I can cross-check them against my saved file for French to see how many of those should be re-add through the Sentence Collector. I will only re-add ones that were not already rejected anyway.

These are duplicates that were, for some reason, not recognized by the Sentence Collector and therefore added multiple times, while the export did recognize it and didn't export it.

**Comment by MichaelKohler** [comment]
This just got a bit more complicated, @rileyjshaw @phirework let's sync up once you have an answer to how we can avoid simple sentence fixes invalidating existing recorded hours (if that is even the case).

**Comment by MichaelKohler** [comment]
# How the export works
As far as I can see the following happens on an import:

* Sentences get read and put into the db from the server/data folder
* Any sentences not used in previous clips and skipped sentences (identification by its ID) get deleted
* All old versions' sentences get marked as `is_used = false`


# Fixing sentences for new recordings
This means that we can first merge ... and then https://github.com/mozilla/voice-web/pull/2262. Once the next deployment to voice.mozilla.org happens, any new sentence clips will use the correct versions.

# What still needs to be done
However the old wrong sentences won't be fully deleted and will still be referenced from the old recordings. Given that we don't want to invalidate the recordings I think that's okay. This means though that we somehow need to change the sentences in the DB as well, to make sure that the resulting export into the dataset get replaced with the correct sentences. For that I'd like to eventually hear the team's thoughts.

**Comment by lissyx** [comment]
FTR, this issue is in the dataset v3.

**Comment by moz-bozden** [comment]
Closing as stale.
